pkg/sql/sem/tree/datum.go

```diff
@@ -2525,6 +2525,10 @@ func AsJSON(d Datum) (json.JSON, error) {
 		return builder.Build(), nil
 	case *DTuple:
 		builder := json.NewObjectBuilder(len(t.D))
+		// We need to make sure that t.typ is initialized before getting the tuple
+		// labels (it is valid for t.typ be left uninitialized when instantiating a
+		// DTuple).
+		t.ResolvedType()
```

Comment:
yeah no. I don't know how this passes the linter. `ResolvedType()` is a method that visually and intendedly returns an attribute of the tuple. I am expected a hard compile error "return value is ignored".

This change needs _at least_ two adjustments:

- ignore the return value explicitly `_ = t.ResolvedType()`
- make the comment explicit: "we are calling this method for its side effects"

Now that is still poor design. `ResolvedType` is morally (even if not practically) a pure function that should not be called here. So the right thing to do is:

1. split the current body of `(*DTuple) ResolvedType()` to place the if + side effects into a _separate_ method without a return value `(*DTuple) maybePopulateTypes()` with no return value
2. call that method from `ResolvedType()`
3. call that method here instead of `ResolvedType()`.